core/templates/dev/head/services/RteHelperService.js

```diff
@@ -181,9 +181,21 @@ oppia.factory('RteHelperService', [
           });
         });
 
-        text = elt.html();
-        text = $filter('removeExtraLines')(text);
-        return text;
+        var textElt = elt[0].childNodes;
```

Comment:
This code should be wrapped in a method and tested with Karma tests covering all the special cases.

Reply:
Done